### PR TITLE
Fix free space issues

### PR DESCRIPTION
**Pull request by jbsparrow**
@NTFSvolume Would you mind testing this fix for me before I merge it? I'm not able to make a ramdisk and whatnot to test it right now 😿

Published to pypi as version `5.6.32.dev1`.

### CODE REVIEW

File: cyberdrop_dl/downloader/downloader.py [hunk 1]
```diff
@@ -196,16 +196,24 @@ async def attempt_task_removal(self, media_item: MediaItem) -> None:
...
     async def download(self, media_item: MediaItem) -> None:
         """Downloads the media item"""
-        if not await self.check_file_can_download(media_item):
-            await self.manager.progress_manager.download_progress.add_skipped()
+        can_download, reason = await self.check_file_can_download(media_item)
+        if not can_download:
+            if reason == 0:
+                await self.manager.progress_manager.download_progress.add_failed()
+            else:
+                await self.manager.progress_manager.download_progress.add_skipped()
```
Comment (NTFSvolume) [inline cyberdrop_dl/downloader/downloader.py:204]: isn't this redundant? the exact same check is done just bellow, inside the try/except block, L211

Comment (jbsparrow) [reply cyberdrop_dl/downloader/downloader.py:204]: Pretty sure you're right... I didn't wanna mess too much with Jules' code that I don't have too much knowledge on 😅

File: cyberdrop_dl/managers/download_manager.py [hunk 2]
```diff
@@ -69,9 +70,12 @@ async def basic_auth(self, username, password) -> str:
...
         return f'Basic {token}'
 
-    async def check_free_space(self) -> bool:
+    async def check_free_space(self, folder: Path = None) -> bool:
         """Checks if there is enough free space on the drive to continue operating"""
-        free_space = shutil.disk_usage(self.manager.path_manager.download_dir.parent).free
+        if folder is None:
+            free_space = disk_usage(self.manager.path_manager.download_dir).free
+        else:
+            free_space = disk_usage(folder).free
```
Comment (NTFSvolume) [inline cyberdrop_dl/managers/download_manager.py:78]: Tested but `check_free_space` errors out if `folder` does not exists.

**Logs**

[downloader.log](https://github.com/user-attachments/files/17091588/downloader.log)

Comment (NTFSvolume) [reply cyberdrop_dl/managers/download_manager.py:78]: Reviewing the code downstream seems that the reason Jules originally used `self.manager.path_manager.download_dir.parent` instead of just `self.manager.path_manager.download_dir` was because at this point in the code `download_dir` has not been created yet.

Comment (jbsparrow) [reply cyberdrop_dl/managers/download_manager.py:78]: Would switching it to `self.manager.path_manager.download_dir.drive` work? I'm not sure if it would work on a ramdisk.

File: cyberdrop_dl/managers/download_manager.py [hunk 3]
```diff
@@ -69,9 +70,12 @@ async def basic_auth(self, username, password) -> str:
...
         return f'Basic {token}'
 
-    async def check_free_space(self) -> bool:
+    async def check_free_space(self, folder: Path = None) -> bool:
         """Checks if there is enough free space on the drive to continue operating"""
-        free_space = shutil.disk_usage(self.manager.path_manager.download_dir.parent).free
+        if folder is None:
+            free_space = disk_usage(self.manager.path_manager.download_dir.drive).free
+        else:
+            free_space = disk_usage(folder).free
```
Comment (NTFSvolume) [inline cyberdrop_dl/managers/download_manager.py:78]: Yeah, `.drive` won't work on unix. It always returns the root. Also, i was wrong about `download_dir`. It's created at `downloader.startup()` which is called inside `crawler.startup()`. 

https://github.com/jbsparrow/CyberDropDownloader/blob/b4dc0021f041cd2574d6b21194d6f18928cd8e76/cyberdrop_dl/downloader/downloader.py#L127

That's before any download and space check.  Using `self.manager.path_manager.download_dir` works just fine. 
The problem is with the parameter `folder`. I was able to fix it with this:

```suggestion
        if folder is None:
            free_space = disk_usage(self.manager.path_manager.download_dir).free
        else:
            folder = folder.resolve()
            while not folder.is_dir() and folder.parents:
                folder = folder.parent   
            # check if we reached an anchor (root) that does not exists, ex: disconnected USB drive
            if not folder.is_dir(): 
                return False
            free_space = disk_usage(folder).free
```
Basically, if `folder` does not exists, traverse it until you find a parent that does and check free space on it

Its works but now there is another minor issue. The download fails but is does not show in the UI summary, only in the logs. UI shows 0 files failed or skipped and no download errors.

**Logs**

[downloader.log](https://github.com/user-attachments/files/17092217/downloader.log)

UI_screenshot

Comment (jbsparrow) [reply cyberdrop_dl/managers/download_manager.py]: Resolved, the files were being added as skipped to the download progress bar but not for the download stats manager.

**Review by NTFSvolume — APPROVED** [review]
Tested. All good

**Review by datawhores — APPROVED** [review]
